libraries/Microsoft.Bot.Builder.Dialogs/DialogContext.cs

```diff
@@ -659,8 +659,7 @@ public async Task<bool> EmitEventAsync(string name, object value = null, bool bu
         /// <returns>A string representing the current locale.</returns>
         public string GetLocale()
         {
-            var locale = ((TurnContext)Context).Locale ?? null;
-            return locale;
+            return Context.TurnState.Get<string>("turn.locale");
```

Comment:
Note that this string is defined in `TurnContext`. However the goal is to deprecate the GetLocale() method in a future PR, so I don't want to make the constant in TurnContext public just to deprecate it. This coupling should never have happened unfortunately :(